crates/bevy_ecs/examples/world_diagnostic.rs

```diff
@@ -0,0 +1,115 @@
+//! In this example, we use a system to print diagnostic information about the world.
+
+#![expect(
+    missing_docs,
+    reason = "Trivial example types do not require documentation."
+)]
+
+use bevy_ecs::prelude::*;
+use bevy_ecs_macros::{ScheduleLabel, SystemSet};
+
+fn empty_system() {}
+
+fn first_system() {}
+
+fn second_system() {}
+
+fn increase_game_state_count(mut state: ResMut<GameState>) {
+    state.counter += 1;
+}
+
+fn sync_counter(state: Res<GameState>, mut query: Query<&mut Counter>) {
+    for mut counter in query.iter_mut() {
+        counter.0 = state.counter;
+    }
+}
+
+#[derive(Resource, Default)]
+struct GameState {
+    counter: usize,
+}
+
+#[derive(SystemSet, Hash, Clone, Copy, PartialEq, Eq, Debug)]
+enum MySet {
+    Set1,
+    Set2,
+}
+
+#[derive(Component)]
+struct Counter(usize);
+
+#[derive(Component)]
+struct Player;
+
+#[derive(Component)]
+#[component(storage = "SparseSet")]
+struct HighlightFlag;
+
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub enum ScheduleLabel {
+    Foo,
+    Bar,
+}
+
+/// A special label for diagnostic.
+/// If the diagnostic system running in a commonly used label, like the ones
+/// defined as `bevy_app::CoreSchedule`, it is not able to get the corresponding
+/// [`Schedule`] instance. ([`World::run_schedule_ref`] for reference)
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct DiagnosticLabel;
+
+/// World diagnostic example.
+/// You can also use [`World::diagnose`] which does not require mutable access.
```

Comment:
If you want you can also write a commented out version of using [`World::diagnose`] as an alternative.